telematic_system/scripts/log_analysis/README.md

```diff
@@ -0,0 +1,119 @@
+- Python environment setup
+    1. Install python
+        ```
+        sudo apt update 
+        sudo apt install python3
+        ```
+    2. Check python version
+        ```
+        python3 --version
+        ```
+        Recommended version is `3.10`
+    3. Create a virtual environment. Navigate to `cda-telematics/telematic_system/scripts/log_analysis` directory, and run below command:
+        ```
+        python3 -m venv .venv
+        ```
+    4. Activate virtual environment.
+        ```
+        .venv\bin\activate
+        ```
+        Note: Need to run this command to activate virtual environment every time openning a new terminal.
+- Install depedencies:
+    - Install debian packages 
+        ```
+        sudo apt install libcairo2-dev libxt-dev libgirepository1.0-dev
+
+        ```
+    - Install python packages
+        ```
+        pip install -r requirements.txt
+        ```
+- Clone repos:
+    - Clone cda-telematics GitHub repos
+    ```
+    git clone https://github.com/usdot-fhwa-stol/cda-telematics.git
+    ```
+
+
+# Process V2xHub bridge log 
+1. Download v2xhub logs to the current folder.
+2. Run command to generate data publishing metrics.
+    ```
+    python3 parse_v2xhub_telematic_plugin_logs.py  --log_file_path <input-file-name>
+
+    e.g:
+    python3 parse_v2xhub_telematic_plugin_logs.py  --log_file_path T20_R6-13_V2XHub.log 
+    ```
+# Process Streets bridge log 
+1. Download streets bridge logs to the current folder.
+2. Run command to generate data publishing metrics.
+    ```
+    python3 parse_streets_bridge_logs.py  <path-to-log-file>
+    ```
+# Process Cloud bridge log 
+1. Download streets bridge logs to the current folder.
+2. Run command to generate data publishing metrics.
+    ```
+    parse_cloud_bridge_logs.py <path-to-log-file>
+
+    e.g:
+    python3 parse_cloud_bridge_logs.py  T20_R6-9_carma_cloud.log 
+    python3 parse_cloud_bridge_logs.py  T20_R10-13_carma_cloud.log 
+    ```
+
+# Process Vehicle bridge log 
+1. Download vehicle bridge logs to the current folder.
+2. Run command to generate data publishing metrics.
+    ```
+    python3 parse_vehicle_bridge_logs.py  <path-to-log-file>
+
+    e.g:
+    python3 parse_vehicle_bridge_logs.py T20_R6_R13_fusion/T20_R6_fusion.log 
+    ```
+# Process Messaging Server log 
+1. Download messaging server logs to the current folder.
+2. Run command to generate data publishing metrics.
+    ```
+    parse_messaging_server_logs.py <path-to-log-file>
+
+    e.g:
+    python3 parse_messaging_server_logs.py T20_R6-13_messaging_server.log
+    ```
+# Metric analysis
+## Latency
+1. Create a folder with the test case name.
+For example, test case 20:
+    ```
+    mkdir T20
+    ```
+2. Copy all the T20_*_messaging_server_*_delay_parsed.csv files to this new folder `T20`
+3. Run plot latency script to generate plots for those csv files with delay metrics in folder `T20`.
+    ```
+    python3 latencyPlotter.py <folder-name or test case name>
```

Comment:
This will depend on where the folder is created, so the argument should be <path to folder..>